### PR TITLE
[MI-3643] Add metrics to show messages confirmed count

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
- Add metrics to show messages confirmed count
- Fixed unit test cases

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -177,6 +179,15 @@ func NewMetrics(info InstanceInfo) Metrics {
 	}, []string{"action", "source", "is_direct", "discarded_reason"})
 	m.registry.MustRegister(m.filesCount)
 
+	m.messagesConfirmedCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemEvents,
+		Name:        "messages_confirmed",
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/metrics/metrics.go:185]: same https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/355#discussion_r1368384826

File: server/handlers/handlers.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ const (
 	discardedReasonMaxFileSizeExceeded    = "max_file_size_exceeded"
 
 	actionSourceMSTeams    = "msteams"
+	ActionSourceMattermost = "mattermost"
```
Comment (lieut-data) [inline server/handlers/handlers.go:50]: This doesn't feel like a constant that `handlers` should define. Can we export this from `metrics` instead?

File: server/metrics/metrics.go [hunk 3]
```diff
@@ -183,6 +185,15 @@ func NewMetrics(info InstanceInfo) Metrics {
...
```
Comment (lieut-data) [inline server/metrics/metrics.go:191]: Can we append `_total` as a suffix? I missed this in the original design document.

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks again, @ayusht2810!

**Review by manojmalik20 — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]

**Pull request merged**